numpy/_build_utils/apple_accelerate.py

```diff
@@ -8,8 +8,13 @@
 
 def uses_accelerate_framework(info):
     """ Returns True if Accelerate framework is used for BLAS/LAPACK """
+    # If we're not building on Darwin (macOS), don't use Accelerate
     if sys.platform != "darwin":
         return False
+    # If we're building on macOS, but targeting a different platform,
+    # don't use Accelerate.
+    if os.getenv('_PYTHON_HOST_PLATFORM', None):
```

Comment:
Should this be `if os.getenv('"_PYTHON_HOST_PLATFORM", None) != "darwin"`? (and ditto below)

In fact, maybe we should have a helper for fetching the build platform?

Reply:
No - you need to check both `sys.platform == 'darwin'` (to confirm that you're compiling on a macOS machine at all), and *then* that you're cross compiling. 

_PYTHON_HOST_PLATFORM will be set to an identifier for the platform you are compiling for (e.g., `"iphoneos.arm64"`) indicating the architecture that you're compiling for. It won't be set at all if you're actually compiling for macOS.

As for whether a helper method is required... sure; I suppose it could be factored out if that's deemed worthwhile.

Reply:
My point was that `_PYTHON_HOST_PLATFORM=darwin` seems like a reasonable case to want to handle too, rather than assuming that this will never happen. (I wasn't suggesting folding the two checks into one.)

Reply:
Setting `_PYTHON_HOST_PLATFORM='darwin'` would make no sense - because you'd be telling distutils that you wanted to compile for darwin... but the libraries and headers needed to support a darwin build are only available on macOS... i.e., darwin. If you want to build for darwin, you can just build for darwin - there's no need to cross-compile for the platform you're already on.

Reply:
It would make sense if the API was "this env var gives you the target platform", and cross compilation was detected as `build platform != target platform`, which would be a good and simple API :-). I guess in practice it's likely that other code is already using a check like this and so distutils is stuck with the more complicated rule that if build and target platform are the same then it has to special case that and unset the envvar.

By the way, you should bring up this cross compilation thing on distutils-sig. PEP 517 currently doesn't have any way to specify the target platform.